From c219501dc1079909b56c435cd2622b88ff3f50d0 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Fri, 22 Mar 2019 09:26:30 +0800
Subject: [PATCH 031/706] serial: 8250: ls1028: change baud rate divisor for
 CFP and PXP

needs updated baudrate divisor

Signed-off-by: Sudhanshu Gupta <sudhanshu.gupta@nxp.com>
(cherry picked from commit f203dc3c8f0bc3f9745aad0efc8250ec9a397c68)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/tty/serial/8250/8250_port.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/tty/serial/8250/8250_port.c b/drivers/tty/serial/8250/8250_port.c
index 5cbc13e3d316..a1ffa3b3379f 100644
--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -2635,7 +2635,13 @@ serial8250_do_set_termios(struct uart_port *port, struct ktermios *termios,
 	cval = serial8250_compute_lcr(up, termios->c_cflag);
 
 	baud = serial8250_get_baud_rate(port, termios, old);
+#if defined(CONFIG_EMU_PXP)
+	quot = 0x01;
+#elif defined(CONFIG_EMU_CFP)
+	quot = 0x02;
+#else
 	quot = serial8250_get_divisor(up, baud, &frac);
+#endif
 
 	/*
 	 * Ok, we're now changing the port state.  Do it with
-- 
2.17.1

